src/Estimators/EstimatorManagerNew.cpp

```diff
@@ -48,76 +50,57 @@ EstimatorManagerNew::EstimatorManagerNew(const QMCHamiltonian& ham, Communicate*
-      addMainEstimator(std::move(estimator));
-    else
-      scalar_ests_.push_back(std::move(estimator));
-    return true;
-  }
-  else
-    return false;
-}
-
-using SPOMap = std::map<std::string, const std::unique_ptr<const SPOSet>>;
-
 //initialize the name of the primary estimator
 EstimatorManagerNew::EstimatorManagerNew(Communicate* c,
-                                         EstimatorManagerInput&& emi,
+                                         EstimatorManagerInput emi,
                                          const QMCHamiltonian& H,
                                          const ParticleSet& pset,
                                          const TrialWaveFunction& twf)
     : RecordCount(0), my_comm_(c), max4ascii(8), FieldWidth(20)
 {
   for (auto& est_input : emi.get_estimator_inputs())
   {
-    bool estimator_made = false;
-    estimator_made =
-        estimator_made || createEstimator<SpinDensityInput>(est_input, pset.getLattice(), pset.getSpeciesSet());
-    estimator_made = estimator_made ||
-        createEstimator<MomentumDistributionInput>(est_input, pset.getTotalNum(), pset.getTwist(), pset.getLattice());
-    estimator_made = estimator_made ||
-        createEstimator<OneBodyDensityMatricesInput>(est_input, pset.getLattice(), pset.getSpeciesSet(),
-                                                     twf.getSPOMap(), pset);
-    if (!estimator_made)
+    if (has<SpinDensityInput>(est_input))
+      operator_ests_.push_back(std::make_unique<SpinDensityNew>(std::move(std::get<SpinDensityInput>(est_input)),
+                                                                pset.getLattice(), pset.getSpeciesSet()));
+    else if (has<MomentumDistributionInput>(est_input))
+      operator_ests_.push_back(
+          std::make_unique<MomentumDistribution>(std::move(std::get<MomentumDistributionInput>(est_input)),
+                                                 pset.getTotalNum(), pset.getTwist(), pset.getLattice()));
+    else if (has<OneBodyDensityMatricesInput>(est_input))
+      operator_ests_.push_back(
+          std::make_unique<OneBodyDensityMatrices>(std::move(std::get<OneBodyDensityMatricesInput>(est_input)),
+                                                   pset.getLattice(), pset.getSpeciesSet(), twf.getSPOMap(), pset));
+    else
       throw UniformCommunicateError(std::string(error_tag_) +
                                     "cannot construct an estimator from estimator input object.");
   }
+
   for (auto& scalar_input : emi.get_scalar_estimator_inputs())
   {
-    bool estimator_made = false;
-    estimator_made      = estimator_made || createScalarEstimator<LocalEnergyInput>(scalar_input, H);
-    estimator_made      = estimator_made || createScalarEstimator<CSLocalEnergyInput>(scalar_input, H);
-    estimator_made      = estimator_made || createScalarEstimator<RMCLocalEnergyInput>(scalar_input, H);
-    if (!estimator_made)
+    std::unique_ptr<ScalarEstimatorBase> scalar_est;
```

Comment:
This would be better written like this
```c++
   // since we can count on these being scalar estimator inputs and the existence of each inputs Consumer type
   // we don't need to check for no type being made only deal with the monotype, which if it shows up here inidicates
   // a developer error.

    bool estimator_made = std::visit(
                                     [this, &H](auto scalar_input) {
          using T = std::decay_t<decltype(scalar_input)>;
          if constexpr (std::is_same_v<T, std::monostate>)
            return false;
          else
          {
            auto estimator = std::make_unique<typename decltype(scalar_input)::Consumer>(std::move(scalar_input), H);
            if (estimator->isMainEstimator())
              addMainEstimator(std::move(estimator));
            else
              scalar_ests_.push_back(std::move(estimator));
          }
          return true;
        },
        scalar_input);

    if (!estimator_made)
      throw UniformCommunicateError(std::string(error_tag_) +
                                    "cannot construct a scalar estimator from scalar estimator input object.");